docs/advanced.rst

```diff
@@ -246,6 +248,73 @@ You can add custom transports using the attribute ``rtps.userTransports``.
 
 All Transport configuration options can be found in the section :ref:`transportdescriptors`.
 
+.. _comm-transports-shm:
+
+Shared memory Transport (SHM)
+=============================
+
+The shared memory transport enables fast communications between entities running in the same processing unit/machine,
+relying on the shared memory mechanisms provided by the host operating system.
+
+SHM transport provides better performance than other transports like UDP / TCP, even when these transports use loopback
+interface. This is mainly due to the following reasons:
+
+ * Large message support: Network protocols need to fragment data in order to comply with the specific protocol and
+   network stacks requirements.
+   SHM transport allows the copy of full messages where the only size limit is the machine's memory capacity.
+
+ * Reduce the number of memory copies: When sending the same message to different endpoints, SHM transport can
+   directly share the same memory buffer with all the destination endpoints.
+   Other protocols require to perform one copy of the message per endpoint.
+
+ * Less operating system overhead: Once initial setup is completed, shared memory transfers require much less system
+   calls than the other protocols.
+   Therefore there is a performance/time consume gain by using SHM.
+
+When two participants on the same machine have SHM transport enabled, all communications between them are automatically
+performed by SHM transport only.
+The rest of the enabled transports are not used between those two participants.
+
+To enable SHM transport in a Participant, you need to add the SharedMemTransportDescriptor to the
+``rtps.userTransports`` attribute (C++ code) or define a transport_descriptor of type SHM in the
+XML file (see below examples).
```

Comment:
This will not be the case anymore, as SHM will be active by default. We may mention how to proceed if the user wants to change the default parameters of the transport, though.

Reply:
you're so right